Peer.Domain/Formatters/CompactFormatter.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Peer.Domain.Util;
+
+namespace Peer.Domain.Formatters;
+
+//q(cn): Is there just an object that represents the parts of the format? 
+//  if we're talking format it's include/exclude different properties and max
+//  widths...
+//var x = new
+//{
+//    Id = ("Id", 0, 4, true),
+//    Title = ("Title", 1, 40, true),
+//    Status = ("S", 2, 1, true),
+//    Url = ("Url", 3, 35, true)
+//}; or similar sort of structure?
+public class CompactFormatter : IPullRequestFormatter
+{
+    private readonly ISymbolProvider _symbolProvider;
+    private const char _ellipsis = '\u2026';
+    private static readonly string _header = $"{"Id",-4} {"Title",-40} {"\ud83d\udea7"} Url";
+    public CompactFormatter(ISymbolProvider symbolProvider)
+    {
+        _symbolProvider = symbolProvider;
+    }
+
+    public IEnumerable<string> FormatLines(IEnumerable<PullRequest> pullRequests)
+    {
+        Validators.ArgIsNotNull(pullRequests, nameof(pullRequests));
+
+        yield return _header;
+
+        foreach (var pullRequest in pullRequests)
+        {
+            yield return CreatePullRequestLine(pullRequest);
+        }
+    }
+
+    private string CreatePullRequestLine(PullRequest pr)
+    {
+        var id = PadOrTruncate(pr.Id, 4);
+        var title = PadOrTruncate(pr.Descriptor.Title, 40);
+        var status = _symbolProvider.GetSymbol(pr.State.Status);
+
+        return $"{id} {title} {status} {pr.Url}";
+    }
+
+    private string PadOrTruncate(string value, int length)
+    {
+        return value.EnumerateRunes().Count() > length
+            ? value[0..(length - 1)] + _ellipsis
```

Comment:
Does the slice syntax already respect runes or should you use  value.EnumerateRunes().Take( )?

Reply:
I actually don't think I need this I had it in while I was debugging because I thought it would solve some alignment issues I was having g but I think I was wrong

Reply:
Ok yea I thought string operations already worked as intended so it was just the mismatch that stood out to me